browser/ui/webui/sanitized_image_source_unittest.cc

```diff
@@ -79,35 +91,48 @@ class SanitizedImageSourceTest : public testing::Test {
   network::TestURLLoaderFactory test_url_loader_factory_;
 
   TestingProfile profile_;
-  SanitizedImageSource source_;
+  BraveSanitizedImageSource source_;
 };
 
-TEST_F(SanitizedImageSourceTest, ImageIsDecoded) {
+TEST_F(BraveSanitizedImageSourceTest, ImageIsDecoded) {
   auto url = GURL("https://example.com/image.png");
   EXPECT_TRUE(Decode(url, kValidImage));
 }
 
-TEST_F(SanitizedImageSourceTest, PaddedImageIsDecoded) {
+TEST_F(BraveSanitizedImageSourceTest, ImageIsDecodedAndResized) {
+  // original size is 204x212
+  auto url = GURL("https://example.com/image.png");
+  auto result = Decode(url, kValidImage, "10x20");
+  EXPECT_TRUE(result);
+  // Check the result is PNG resized to 19x20 (to be able cut into 10x20)
+  auto bitmap = gfx::PNGCodec::Decode(base::as_byte_span(*result),
```

Comment:
Hey this doesn't seem right - the image should be resized to fit into a 10x20 box, so it should probably be something like `10x12` (just guestimating)

Reply:
in fact on NTP we also crop the image: So if we ask 10x20 the right answer is 19x20: the minimal size that can be cropped to 10x20.
Resizing it to 10x12 results in a) having a heigh gap (20-12) or b) upscaling to 19x20, then cropping.

Reply:
realised this further down :smile: